### PR TITLE
Add ZoneSync method #52

### PR DESCRIPTION
**Pull request by olimpias**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions.

### Description of the Change

Solves #52
According to [syncZone Api](https://www.vinyldns.io/api/sync-zone.html), the http call support is added.

### Why Should This Be In The Package?

people might need it.

### Benefits

Adds a value to project  with sync zone.

### Verification Process

unit tests are added to check functionality.

### CODE REVIEW

File: vinyldns/zones_test.go [hunk 1]
```diff
@@ -554,6 +554,42 @@ func TestZoneChanges(t *testing.T) {
 	}
 }
 
+func TestZoneSync(t *testing.T) {
+	zoneChangesJSON, err := readFile("test-fixtures/zones/zone-sync.json")
```
Comment (mdb) [inline vinyldns/zones_test.go]: Can we call this `zoneSyncJSON`, just for consistency's sake?

File: vinyldns/groups_test.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ func TestGroupCreate(t *testing.T) {
 		t.Error(err)
 	}
 	server, client := testTools([]testToolsConfig{
-		testToolsConfig{
```
Comment (mdb) [inline vinyldns/groups_test.go]: I know it's a bit redundant, but how would you feel about removing `testToolsConfig` from _all_ `*_test.go` files, but in a _separate_ PR? I'm a bit hesitant to make `groups_test.go`'s coding style unique, while the other `*_test.go` files explicitly declare a `testToolsConfig{}`. Please let me know if you disagree.

File: vinyldns/endpoints_test.go [hunk 3]
```diff
@@ -87,6 +87,17 @@ func TestZoneEP(t *testing.T) {
...
 }
 
+func TestZoneSyncEP(t *testing.T) {
+	zone := zoneSyncEP(c, "123")
+	expected := "http://host.com/zones/123/sync"
+
+	if zone != expected {
+		fmt.Printf("Expected: %s", expected)
+		fmt.Printf("Actual: %s", zone)
+		t.Error("zoneEP should return the right endpoint")
```
Comment (mdb) [inline vinyldns/endpoints_test.go]: I believe the error should read:

```golang
t.Error("zoneSyncEP should return the right endpoint")
```

File: go.sum [hunk 4]
```diff
@@ -1,4 +1,8 @@
 github.com/gobs/pretty v0.0.0-20180724170744-09732c25a95b h1:/vQ+oYKu+JoyaMPDsv5FzwuL2wwWBgBbtj/YLCi4LuA=
 github.com/gobs/pretty v0.0.0-20180724170744-09732c25a95b/go.mod h1:Xo4aNUOrJnVruqWQJBtW6+bTBDTniY8yZum5rF3b5jw=
+github.com/smartystreets/assertions v1.0.1 h1:voD4ITNjPL5jjBfgR/r8fPIIBrliWrWHeiJApdr3r4w=
+github.com/smartystreets/assertions v1.0.1/go.mod h1:kHHU4qYBaI3q23Pp3VPrmWhuIUrLW/7eUrw0BU5VaoM=
 github.com/smartystreets/go-aws-auth v0.0.0-20180515143844-0c1422d1fdb9 h1:hp2CYQUINdZMHdvTdXtPOY2ainKl4IoMcpAXEf2xj3Q=
 github.com/smartystreets/go-aws-auth v0.0.0-20180515143844-0c1422d1fdb9/go.mod h1:SnhjPscd9TpLiy1LpzGSKh3bXCfxxXuqd9xmQJy3slM=
+github.com/smartystreets/gunit v1.0.4 h1:tpTjnuH7MLlqhoD21vRoMZbMIi5GmBsAJDFyF67GhZA=
+github.com/smartystreets/gunit v1.0.4/go.mod h1:EH5qMBab2UclzXUcpR8b93eHsIlp9u+pDQIRp5DZNzQ=
```
Comment (mdb) [inline go.sum]: Is there a reason why `github.com/smartystreets/gunit` has been added as a dependency? I'm a bit confused about this.

File: vinyldns/groups_test.go (hunk 2 above)
Comment (olimpias) [reply vinyldns/groups_test.go]: Sure its fine.

File: go.sum (hunk 4 above)
Comment (olimpias) [reply go.sum]: When I run `go mod tidy`, those dependencies are added.

Comment (mdb) [reply go.sum]: Thanks for clarifying. This is true when I run `go mod tidy` as well; appreciate it!

Comment (olimpias) [reply go.sum]: I dont know why but even I deleted them, it worked :)

Comment (olimpias) [reply go.sum]: In my update, I removed this changes. If you want, I can add them back.

**Pull request merged**